src/python/pants/core/goals/test.py

```diff
@@ -355,6 +355,26 @@ def activated(cls, union_membership: UnionMembership) -> bool:
             """
         ),
     )
+    shard = StrOption(
+        "--shard",
+        default="",
+        help=softwrap(
+            """
+            A shard specification of the form "k/N", where N is a positive integer and k is a
+            non-negative integer less than N.
```

Comment:
This is terse, but maybe could be more accessible?
```suggestion
            A shard specification of the form "k/N". N is a positive integer, and k is either 0
            or a positive integer less than N.
```

Reply:
Thanks for the suggestions, but TBH I like my phrasing better :)

Reply:
I know it's merged already, but to me, zero based counting is nonsensical to me in this context. More natural is to say, shard 1, shard 2, ... so in that case it would be `.. k is a positive integer less than or equal to N.`.

Reply:
Bigggg +1. This always confused me w/ v1 sharding. I think `1/3`, `2/3`, and `3/3` makes much more intuitive sense to me.

It's not too late to change this - a release hasn't gone out yet.

Reply:
And most important, that the last shard is not `--shard 2/3` !

Reply:
There is prior art here, and I've always encountered shard numbering as zero-based. Presumably because shards are computing by mod-hashing a string, and X mod N is always a number 0<=m<N.  

So to not use zero-based numbering would be artificial.

Reply:
Yeah I can see how 2/3 being the last shard looks weird. If I had done two options (--shard and --num-shards) it sounds like that would have looked less weird?

Reply:
Which I considered, but felt like this was more streamlined.

Reply:
My 1/100c I liked kaos' page number example.

In either case I think people will stumble but ultimately I think we should appease the reader and not the pipeline author.

There will be more people reading the pipeline info (who might not be engineers) than people authoring pipelines.

Reply:
> I find it equally artificial to restrict the representation of the interface of the parameter to the algorithm used to calculate the shard numbers; and also, how well versed in the prior art of sharding is expected in the target audience using this?

Big +1. My gut says pretty strongly that `3/3` is much more intuitive. Even after taking my discrete math classes etc, I still feel that way. And many of our users won't have taken discrete math etc

But I too will leave this to you @benjyw

Reply:
Technically we can support both by letting 3/3 wrap around and just changing the validation...

Reply:
> Technically we can support both

That sounds confusing. Generally w/ Pants we've tried to avoid having multiple ways of doing the same thing

Reply:
Yeah, fair. I see the arguments, but TBH I'm more inclined on balance to leave this be, as pretty much any sharding scheme anyone has ever encountered is zero-based, and it's covered in the help message, and if you set it wrong you get a helpful error message. So the consequences are pretty minor either way.

Reply:
> as pretty much any sharding scheme anyone has ever encountered is zero-based

Maybe my memory is fooling me, but I don't think this has been my experience. 

Or perhaps what I'm remembering is whenever I "shard" in my daily life, like how to split up # of worksheets to hand out to students. I would never have used 2/3 to represent the final group hah

I definitely encourage making the change: we want Pants to be highly intuitive